pandas/tests/frame/test_nonunique_indexes.py

```diff
@@ -416,9 +416,8 @@ def test_columns_with_dups(self):
             [[1, 2, 1., 2., 3., 'foo', 'bar']], columns=list('ABCDEFG'))
         assert_frame_equal(df, expected)
 
-        # this is an error because we cannot disambiguate the dup columns
-        pytest.raises(Exception, lambda x: DataFrame(
-            [[1, 2, 'foo', 'bar']], columns=['a', 'a', 'a', 'a']))
+        # TODO: Is this an error because we cannot disambiguate the dupes?
+        DataFrame([[1, 2, 'foo', 'bar']], columns=['a', 'a', 'a', 'a'])
```

Comment:
@jreback : The test was setup incorrectly, so I'm not sure this bug was ever fixed properly...

(if that was the intention)

Reply:
hmm, did this raise before? that is weird

Reply:
Yes, it raised because the `lambda` has a variable in its signature.  However, you can see that the function call to `pytest.raises` provides no arguments to the `lambda`.

Hence, the `Exception` we were getting was really a `TypeError` due to insufficient arguments to the `lambda` call.

Reply:
ok, then i guess just remove this  (or assert the correct creation)

Reply:
Based on what I see around this test, it seems like we're supposed to mangle the columns.  However, the output of this `DataFrame` construction does not mangle, which looks like a bug.

Perhaps we could do a separate xfail test for that instead?

Reply:
that's fine too

Reply:
Actually, never mind.  I misread the test.  I think I'll just update with expected output similar to what's done in the rest of the test.